### ISSUE TITLE
Custom log handlers

### ISSUE THREAD

**Issue by JanBeelte** [chrischtel/nexlog#10]
Hello,

I am just playing around with nexlog and really like the clean and thread safe design, thanks for the great work!
In my application I am planning to show the log output in a UI tab in parallel to the console output.
Is it possible to attach my own handler to update my UI every time a new log message comes in?

Best,

Jan

**Comment by chrischtel** [comment]
Hi Jan,

Thank you for your interest in **nexlog**!

Yes, implementing a UI-based log handler is definitely possible. Could you let me know which branch of **nexlog** you're working on—`develop` or `stable`?

Here's a quick concept I put together. Does this align with what you had in mind?

```zig
fn uiWrite(level: LogLevel, message: []const u8, metadata: ?LogMetadata) !void {
    // Append the log message to the UI
    _ = metadata; // Metadata can be processed if needed
    my_ui.appendLog(level, message);
}

fn uiFlush() !void {
    // Optional: Add behavior to flush UI logs if necessary
}

fn uiDeinit() void {
    // Optional: Add cleanup logic for UI resources
}

// Initialize and register the UI log handler
const ui_handler = LogHandler.init(uiWrite, uiFlush, uiDeinit);
try logger.registerHandler("ui", ui_handler);
```
Best regards, 
Christian

**Comment by chrischtel** [comment]
Hello again,

I’ve got a prototype working already. I’ll do some more testing, improve the documentation, and make some optimizations. I’m planning to release version 0.1.0-beta.1 tomorrow. After that, you can try out the new LogHandler interface and let me know about any bugs, edge cases, or things you don’t like.

**Comment by JanBeelte** [comment]
Wow great, thanks for the quick response, looking forward to trying it out :-)

**Comment by chrischtel** [comment]
Thank you! I merged into the develop branch. You can try it out!

**Comment by JanBeelte** [comment]
Nice Christmas present, thanks a lot!
I will try it out as soon as I have some time 😃

**Comment by chrischtel** [comment]
So, have you tried it already? I am waiting for feedback to fix any bugs or issues that might exist. You can checkout the v0.2.0-alpha.1 release or the master branch.

**Comment by chrischtel** [comment]
When you experience any problems, just create a new issue.
